### DISCUSSION TITLE
Nested Menu Safe Area

### DISCUSSION THREAD

**Discussion by BrianHung** [radix-ui/primitives#1219] [Ideas] [answered]
It would be nice if components that supported nested menus like dropdowns or context menus also supported 

> a safe space where the mouse could move without triggering the sub-menu to close

https://medium.com/linear-app/invisible-details-2ca718b41a44

There's also a gist at the bottom of the article: https://gist.github.com/eldh/51e3825b7aa55694f2a5ffa5f7de8a6a

**Comment by jjenzz** [comment, marked as answer]
Do you mean this?

![CleanShot 2022-03-03 at 10 14 09](https://user-images.githubusercontent.com/175330/156544192-ad540d46-f129-4e7a-9a21-ec375b28e8a8.gif)

I agree the pointer grace polygon could maybe be improved to follow the mouse like the linear example though. I believe the radix one is a fixed polygon from your initial pointer position but could be wrong there.

**Reply by BrianHung** [reply]
Yep, some type of pointer grace when hovering between an item and its nested menu. 
![image](https://user-images.githubusercontent.com/15721634/156611662-870679b7-1fe0-4339-9794-6443e1400d56.png)

**Reply by jjenzz** [reply]
the aforementioned components [already have a pointer grace](https://github.com/radix-ui/primitives/blob/main/packages/react/menu/src/Menu.tsx#L1246), as demonstrated in my gif above. there is potentially room for improvement though.

**Reply by BrianHung** [reply]
That's good to see in code 😅.  Think my cursor movements are a bit too slow for the 300ms timeout.
